addons/pos_gift_card/static/src/js/GiftCardPopup.js

```diff
@@ -42,35 +44,38 @@ odoo.define("pos_gift_card.GiftCardPopup", function (require) {
         this.env.pos.db.product_by_id[
           this.env.pos.config.gift_card_product_id[0]
         ];
-      this.env.pos.get_order().add_product(gift, {
-        price: this.state.amountToSet,
-        quantity: 1,
-        merge: false,
-        generated_gift_card_ids: giftCard ? giftCard.id : false,
-      });
+
+      let can_be_sold = !giftCard.buy_pos_order_line_id && !giftCard.buy_line_id;
+
+      if (can_be_sold || !giftCard) {
```

Comment:
Why is ther !giftCard ? it means that when you generate a new one, it won't work. more over in the generated_gift_card_ids, you have "giftCard ? giftCard.id : false,"

Reply:
It was working with it but it was actually not needed. When the gift card is False `can_be_sold` will always be true so the `!giftCard` was not needed.